### ISSUE TITLE
Incorrect parse errors when editing

### ISSUE THREAD

**Issue by DomantasJ** [rust-analyzer/rust-analyzer#5662]
Reproduction steps:

1. `cargo new --lib repro`
2. Add `src/a.rs` with the following code:
    ```rust
    fn foo() {
        let a = 1;
    }
    ```
3. Open `repro` in vscode, open `src/a.rs` file.
4. Add an underscore before `a`:
    ```rust
    fn foo() {
        let _a = 1;
    }
    ```
5. Get `Syntax Error: expected SEMICOLON` on `_a` token.

Closing and reopening the file fixes the error, but further edits might break it again.

Here's the syntax tree after adding the underscore, rust analyzer thinks that underscore was added before the space:

```
SOURCE_FILE@0..29
  FN@0..28
    FN_KW@0..2 "fn"
    WHITESPACE@2..3 " "
    NAME@3..6
      IDENT@3..6 "foo"
    PARAM_LIST@6..8
      L_PAREN@6..7 "("
      R_PAREN@7..8 ")"
    WHITESPACE@8..9 " "
    BLOCK_EXPR@9..28
      L_CURLY@9..10 "{"
      WHITESPACE@10..15 "\n    "
      EXPR_STMT@15..19
        PATH_EXPR@15..19
          PATH@15..19
            PATH_SEGMENT@15..19
              NAME_REF@15..19
                IDENT@15..19 "let_"
      WHITESPACE@19..20 " "
      EXPR_STMT@20..26
        BIN_EXPR@20..25
          PATH_EXPR@20..21
            PATH@20..21
              PATH_SEGMENT@20..21
                NAME_REF@20..21
                  IDENT@20..21 "a"
          WHITESPACE@21..22 " "
          EQ@22..23 "="
          WHITESPACE@23..24 " "
          LITERAL@24..25
            INT_NUMBER@24..25 "1"
        SEMICOLON@25..26 ";"
      WHITESPACE@26..27 "\n"
      R_CURLY@27..28 "}"
  WHITESPACE@28..29 "\n"
```

vs code extension version: v0.2.264
rust-analyzer version: 2020-08-03 (e96bfd8)

**Comment by lnicola** [comment]
Can you try the latest nightly or a source build? CC #5659.

**Comment by vallentin** [comment]
@lnicola I just got the same issue, and found this thread.

I just tested using the [latest release on the CI](https://github.com/rust-analyzer/rust-analyzer/actions/runs/194059974) and I can confirm that it solved the issue.

**Comment by DomantasJ** [comment]
Switched to nightly update channel, now it works correctly for me too.
